LambdaEngine/Source/Input/API/Input.cpp

```diff
@@ -7,11 +7,12 @@
 
 namespace LambdaEngine
 {
-	KeyboardState		Input::s_KeyboardStates[2];
-	MouseState			Input::s_MouseStates[2];
-	std::atomic_bool	Input::s_InputEnabled = true;
-	SpinLock			Input::s_WriteBufferLockMouse;
-	SpinLock			Input::s_WriteBufferLockKeyboard;
+	KeyboardState			Input::s_KeyboardStates[2][2];
+	MouseState				Input::s_MouseStates[2][2];
+	std::atomic_bool		Input::s_InputEnabled = true;
+	SpinLock				Input::s_WriteBufferLockMouse;
+	SpinLock				Input::s_WriteBufferLockKeyboard;
+	std::stack<InputMode>	Input::s_InputModeStack;
```

Comment:
Why use a stack instead of a simple bool? Is this in case we will have more than 2 modes?

Reply:
Exactly, stack is for when you for instance open the console window and then open render graph from there. This will mean we have two open GUI windows, so when we close the console window we don't want the state to change. By using the stack it will simply pop one inputMode but keeping one so that the inputMode is as expected.